### ISSUE TITLE
Server API for plugins

### ISSUE THREAD

**Issue by VladimirKalachikhin** [SignalK/charts-plugin#31]
The plugin does not support Server API for plugins.
So, there is no way to make another server plugin use same `/signalk/v1/api/resources/charts/` path.

**Comment by tkurki** [comment]
That is absolutely correct. We really should change the resources/charts api to work like the rest of the resources v2 api: there can be multiple providers for the same resource type.

@panaaj any thoughts? My hunch is that  it should be pretty straightforward. I could get around to doing that during this winter, unless somebody beats me to it.

Do we need anything on the server side, or is it just changing the plugin? Will the multiple providers mechanism be available only in the v2 api?

**Comment by panaaj** [comment]
It does / should support multiple providers on the V2 endpoint (as this leverages the resources API).

For v1 endpoints to support multiple providers it will need both server and plugin. 

My question would be why? Shouldn't we be looking ahead and encouraging the move to V2? 
IMHO, being able to take advantage of this feature is a reason to migrate.

**Comment by VladimirKalachikhin** [comment]
I think that until you have completely banned API v.1, you should require a plugin to comply with all the rules of API v.1. 
The behavior of the Chart plugin is completely inconsistent with the API v.1 documentation. Therefore, if you want to look ahead, you need to disable v.1 path maintenance by the Chart plugin.

**Comment by panaaj** [comment]
> The behavior of the Chart plugin is completely inconsistent with the API v.1 documentation.

This is a separate issue to supporting multiple providers. Fixing this is likely to be a breaking change for clients.

So if we require clients to be updated, that effort should be to support V2 not a breaking change to v1.
